File: Samples-Java/Samples/src/main/java/fragments/dataform/DataFormSchemaSetupFragment.java
package fragments.dataform;


import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.telerik.android.sdk.R;
import com.telerik.widget.dataform.engine.DataFormMetadata;
import com.telerik.widget.dataform.visualization.RadDataForm;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;
import java.io.StringWriter;
import java.io.Writer;

import activities.ExampleFragment;

public class DataFormSchemaSetupFragment extends Fragment implements ExampleFragment {

    @Override
    public String title() {
        return "Schema setup";
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        ViewGroup layoutRoot = (ViewGroup) inflater.inflate(R.layout.fragment_data_form_schema_setup, container, false);
        RadDataForm dataForm = new RadDataForm(this.getActivity());

        String json = loadJSONFromAsset(R.raw.person_extended);

        try {
            JSONObject jsonObject = new JSONObject(json);
            dataForm.setEntity(jsonObject);

            String schema = loadJSONFromAsset(R.raw.person_schema);
            JSONObject jsonSchema = new JSONObject(schema);
            DataFormMetadata metadata = new DataFormMetadata(jsonSchema);
            dataForm.setMetadata(metadata);
        } catch (JSONException e) {
            Log.e("json", "error parsing json", e);
        }

        layoutRoot.addView(dataForm);

        return layoutRoot;

    }

    public String loadJSONFromAsset(int asset) {
        String json;
        try {
            InputStream is = getResources().openRawResource(asset);
            Writer writer = new StringWriter();
            char[] buffer = new char[1024];
            try {
                Reader reader = new BufferedReader(new InputStreamReader(is, "UTF-8"));
                int n;
                while ((n = reader.read(buffer)) != -1) {
                    writer.write(buffer, 0, n);
                }
            } finally {
                is.close();
            }

            json = writer.toString();
        } catch (IOException ex) {
            ex.printStackTrace();
            return null;
        }
        return json;
    }


}

File: Samples-Xamarin/Fragments/Dataform/DataFormSchemaSetupFragment.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;
using System.IO;
using Com.Telerik.Widget.Dataform.Visualization;
using Org.Json;
using Com.Telerik.Widget.Dataform.Engine;
using Android.Support.V4.App;

namespace Samples
{
	public class DataFormSchemaSetupFragment : Fragment, ExampleFragment
	{
		public string Title() {
			return "Json Schema setup";
		}

		public override void OnCreate (Bundle savedInstanceState)
		{
			base.OnCreate (savedInstanceState);

			// Create your fragment here
		}

		public override View OnCreateView (LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
		{
			ViewGroup layoutRoot = (ViewGroup)inflater.Inflate(Resource.Layout.fragment_data_form_schema_setup, container, false);
			RadDataForm dataForm = (RadDataForm)layoutRoot.FindViewById(Resource.Id.data_form_schema);

			String json = LoadJSONFromAsset ("PersonExtended.json");
			try {
				JSONObject jsonObject = new JSONObject (json);
				dataForm.SetEntity(jsonObject);

				String schema = LoadJSONFromAsset("PersonSchema.json");
				JSONObject jsonSchema = new JSONObject(schema);
				DataFormMetadata metadata = new DataFormMetadata(jsonSchema);
				dataForm.Metadata = metadata;
			} catch(JSONException e) {
				Log.Error ("json", "error parsing json", e);
			}

			return layoutRoot;
		}

		private String LoadJSONFromAsset(String fileName)
		{
			String json = null;
			try
			{
				using (StreamReader sr = new StreamReader (Activity.Assets.Open (fileName)))
				{
					json = sr.ReadToEnd ();
				}
			}
			catch (IOException ex)
			{
				Log.Error("error", ex.StackTrace);
			}
			return json;
		}
	}
}

